frontend/src/metabase/nav/containers/MainNavbar/MainNavbarView.tsx

```diff
@@ -85,18 +87,32 @@ function MainNavbarView({
   return (
     <SidebarContentRoot>
       <div>
+        <SidebarSection>
+          <ul>
+            <HomePageLink
+              isSelected={isNonEntityLinkSelected}
```

Comment:
We also need to check `selectedItem.url` for `isSelected` prop. Right now if you open "Browse data", "Home" will also get highlighted